File: video-club-rental/typescript/tests/FixedClock.ts
import type { Clock } from '../src/Clock.js';

const MS_PER_DAY = 86_400_000;

export class FixedClock implements Clock {
  private _today: Date;
  constructor(today: Date) { this._today = today; }
  today(): Date { return this._today; }
  advanceTo(d: Date): void { this._today = d; }
  advanceDays(days: number): void {
    this._today = new Date(this._today.getTime() + days * MS_PER_DAY);
  }
}

File: video-club-rental/python/tests/fixed_clock.py
from __future__ import annotations

from datetime import date, timedelta


class FixedClock:
    def __init__(self, today: date) -> None:
        self._today = today

    def today(self) -> date:
        return self._today

    def advance_to(self, d: date) -> None:
        self._today = d

    def advance_days(self, days: int) -> None:
        self._today = self._today + timedelta(days=days)
